Altered slightly from upstream counterparts in order to make them apply to
the dist tarball (updated context and filenames to match the dist).

https://github.com/PowerDNS/pdns/commit/f407319cb7374baf06698786f4f39d431a36a3cc
https://github.com/PowerDNS/pdns/commit/d8b4ea24dc5dbe7c53c24bfffba24bcae7f58e02

From d8b4ea24dc5dbe7c53c24bfffba24bcae7f58e02 Mon Sep 17 00:00:00 2001
From: Peter van Dijk <peter.van.dijk@powerdns.com>
Date: Mon, 23 Jan 2023 14:54:09 +0100
Subject: [PATCH] lock.hh: include <stdexcept>

---
 pdns/lock.hh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pdns/lock.hh b/pdns/lock.hh
index e8bd82988da..a08951a266d 100644
--- a/lock.hh
+++ b/lock.hh
@@ -22,6 +22,7 @@
 #pragma once
 #include <mutex>
 #include <shared_mutex>
+#include <stdexcept>
 
 /*
   This file provides several features around locks:
From f407319cb7374baf06698786f4f39d431a36a3cc Mon Sep 17 00:00:00 2001
From: Sander Hoentjen <shoentjen@antagonist.nl>
Date: Mon, 20 Feb 2023 16:51:07 +0100
Subject: [PATCH] dnsdist-protocols.hh: include <cstdint>

This fixes building dnsdist with gcc13:
```
In file included from dnsdist-protocols.cc:26:
dnsdist-protocols.hh:32:8: error: use of enum 'typeenum' without previous declaration
   32 |   enum typeenum : uint8_t
      |        ^~~~~~~~
dnsdist-protocols.hh:32:19: error: 'uint8_t' was not declared in this scope
   32 |   enum typeenum : uint8_t
      |                   ^~~~~~~
dnsdist-protocols.hh:25:1: note: 'uint8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
   24 | #include <vector>
  +++ |+#include <cstdint>
   25 | #include <string>
```
---
 pdns/dnsdist-protocols.hh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pdns/dnsdist-protocols.hh b/pdns/dnsdist-protocols.hh
index c5c5743908d..bd2a4bb8ad1 100644
--- a/dnsdist-protocols.hh
+++ b/dnsdist-protocols.hh
@@ -22,6 +22,7 @@
 #pragma once
 
 #include <vector>
+#include <cstdint>
 #include <string>
 
 namespace dnsdist
